contracts/finance/andromeda-weighted-distribution-splitter/src/contract.rs

```diff
@@ -0,0 +1,938 @@
+        info,
+        BaseInstantiateMsg {
+            ado_type: "weighted-splitter".to_string(),
+            operators: None,
+            modules: msg.modules,
+            primitive_contract: None,
+        },
+    )
+}
+
+#[entry_point]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+
+    // Do this before the hooks get fired off to ensure that there is no conflict with the app
+    // contract not being whitelisted.
+    if let ExecuteMsg::AndrReceive(AndromedaMsg::UpdateAppContract { address }) = msg {
+        let splitter = SPLITTER.load(deps.storage)?;
+        let mut andr_addresses: Vec<AndrAddress> = vec![];
+        for recipient in splitter.recipients {
+            if let Recipient::ADO(ado_recipient) = recipient.recipient {
+                andr_addresses.push(ado_recipient.address);
+            }
+        }
+        return contract.execute_update_app_contract(deps, info, address, Some(andr_addresses));
+    };
+
+    contract.module_hook::<Response>(
+        deps.storage,
+        deps.api,
+        deps.querier,
+        AndromedaHook::OnExecute {
+            sender: info.sender.to_string(),
+            payload: encode_binary(&msg)?,
+        },
+    )?;
+
+    match msg {
+        ExecuteMsg::UpdateRecipients { recipients } => {
+            execute_update_recipients(deps, info, recipients)
+        }
+        ExecuteMsg::AddRecipient { recipient } => execute_add_recipient(deps, info, recipient),
+        ExecuteMsg::RemoveRecipient { recipient } => {
+            execute_remove_recipient(deps, info, recipient)
+        }
+        ExecuteMsg::UpdateLock { lock } => execute_update_lock(deps, info, lock),
+
+        ExecuteMsg::Send {} => execute_send(deps, info),
+        ExecuteMsg::AndrReceive(msg) => execute_andromeda(deps, env, info, msg),
+    }
+}
+
+pub fn execute_add_recipient(
+    deps: DepsMut,
+    info: MessageInfo,
+    recipient: AddressWeight,
+) -> Result<Response, ContractError> {
+    // Only the contract's owner can add a recipient
+    require(
+        ADOContract::default().is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {},
+    )?;
+    // No need to send funds
+    require(
+        info.funds.is_empty(),
+        ContractError::FunctionDeclinesFunds {},
+    )?;
+    // Check if splitter is locked
+    let mut splitter = SPLITTER.load(deps.storage)?;
+
+    require(!splitter.locked, ContractError::ContractLocked {})?;
+
+    // Check for duplicate recipients
+
+    let user_index = splitter
+        .recipients
+        .clone()
+        .into_iter()
+        .position(|x| x.recipient == recipient.recipient);
+
+    // If the index exists, the recipient is a duplicate
+    // If the index doesn't exist, it's a new recipient and we may proceed
+    require(user_index == None, ContractError::DuplicateRecipient {})?;
```

Comment:
It looks to me like using `any` is better in this case: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any